SymmetricTree.java

```diff
@@ -0,0 +1,49 @@
+// Time Complexity : O(n), O(n) to generate inorder traversal, O(n) to find root_elem_index in inorder array, O(n) to check if left half is mirror of right half
+// Space Complexity : O(n), O(n) for storing inorder traversal array, O(log n) for stack space
+// Did this code successfully run on Leetcode : no, passed only 191/195 test cases, failed for a case where subtree was not a mirror but inorder traversal left and right halves were mirror
```

Comment:
Nice Idea. I will check this code on leetcode and reach out. Look out for an optimized algorithm in class.